llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -312,6 +312,20 @@ std::vector<StringRef> getKernelNamesUsingAssert(const Module &M) {
   return SPIRKernelNames;
 }
 
+std::vector<StringRef> getKernelNamesUsingAsan(const Module &M) {
+  auto *AsanInitFunction = M.getFunction("__asan_init");
+  if (!AsanInitFunction)
+    return {};
+
+  std::vector<StringRef> SPIRKernelNames;
+  for (const Function &F : M) {
+    if (F.getCallingConv() == CallingConv::SPIR_KERNEL)
+      SPIRKernelNames.push_back(F.getName());
+  }
+
+  return SPIRKernelNames;
```

Comment:
We unconditionally adding *all* kernels to the property and in runtime we only check if the property set is present, regardless of its value. Do we plan to change the latter?

Because if not, then this should not be a separate property set which passes through *all* kernel names for them to be simply completely ignored later. Instead, this should be a single boolean within `SYCL/misc properties`

Reply:
Thanks for your suggestion, I'll investigate this.

Reply:
Okay, I decided to change it to "SYCL/misc properties", Thanks!